File: offer09.cpp
class CQueue {
private:
    stack<int> s1,s2;
public:
    void appendTail(int value) {
        s1.push(value);
    }
    
    int deleteHead() {
        int result = -1;
        if(!s2.empty()){
            result = s2.top();
            s2.pop();
        }
        else if(!s1.empty()){
             while(!s1.empty()){
                s2.push(s1.top());
                s1.pop();
            }
            result = s2.top();
            s2.pop();
        }
        else return -1;
    return result;
    }
};

/**
 * Your CQueue object will be instantiated and called as such:
 * CQueue* obj = new CQueue();
 * obj->appendTail(value);
 * int param_2 = obj->deleteHead();
 */

File: offer09.py
class CQueue(object):

    def __init__(self):
        self.s1, self.s2 = [], []

    def appendTail(self, value: int) -> None:
        self.s1.append(value)


    def deleteHead(self) -> int:
        """
        :rtype: int
        """
        res = -1
        if self.s2:
            res = self.s2.pop()
        elif self.s1:
            while(self.s1):
                self.s2.append(self.s1.pop())
            res = self.s2.pop()
        else: return -1
        return res



# Your CQueue object will be instantiated and called as such:
# obj = CQueue()
# obj.appendTail(value)
# param_2 = obj.deleteHead()
